Sources/BrowserServicesKit/PrivacyConfig/AppPrivacyConfiguration.swift

```diff
@@ -180,28 +188,76 @@ public struct AppPrivacyConfiguration: PrivacyConfiguration {
     }
 
     public func isSubfeatureEnabled(_ subfeature: any PrivacySubfeature,
+                                    cohortID: CohortID?,
                                     versionProvider: AppVersionProvider,
                                     randomizer: (Range<Double>) -> Double) -> Bool {
-
-        switch stateFor(subfeature, versionProvider: versionProvider, randomizer: randomizer) {
+        switch stateFor(subfeature, cohortID: cohortID, versionProvider: versionProvider, randomizer: randomizer) {
         case .enabled:
             return true
         case .disabled:
             return false
         }
     }
 
-    public func stateFor(_ subfeature: any PrivacySubfeature, versionProvider: AppVersionProvider, randomizer: (Range<Double>) -> Double) -> PrivacyConfigurationFeatureState {
+    public func getAllActiveExperiments(versionProvider: AppVersionProvider,
+                                        randomizer: (Range<Double>) -> Double) -> Experiments {
+        Self.experimentManagerQueue.sync {
+               guard let assignedExperiments = experimentManager.experiments else { return [:] }
+               return assignedExperiments.filter { key, value in
+                   stateFor(subfeatureID: key, experimentData: value, versionProvider: versionProvider, randomizer: randomizer) == .enabled
+               }
+           }
+    }
 
-        let parentState = stateFor(featureKey: subfeature.parent, versionProvider: versionProvider)
-        guard case .enabled = parentState else { return parentState }
+    private func stateFor(subfeatureID: SubfeatureID, experimentData: ExperimentData, versionProvider: AppVersionProvider,
+                          randomizer: (Range<Double>) -> Double) -> PrivacyConfigurationFeatureState {
+        guard let parentFeature = PrivacyFeature(rawValue: experimentData.parentID),
+              let subfeatureData = subfeatures(for: parentFeature)[subfeatureID] else {
+            return .disabled(.featureMissing)
+        }
+        return stateFor(parentFeature: parentFeature,
+                        subfeatureData: subfeatureData,
+                        subfeatureID: subfeatureID,
+                        cohortID: experimentData.cohort,
+                        assignCohortEnabled: false,
+                        versionProvider: versionProvider,
+                        randomizer: randomizer)
+    }
+
+    public func stateFor(_ subfeature: any PrivacySubfeature,
+                         cohortID: CohortID?,
+                         versionProvider: AppVersionProvider,
+                         randomizer: (Range<Double>) -> Double) -> PrivacyConfigurationFeatureState {
+        Self.experimentManagerQueue.sync {
```

Comment:
You should limit the sync queue block to only what truly requires synchronization. Avoid including any dependencies inside it (!). For example, the InternalUserDecider should handle its responsibilities elsewhere because if someone adds main queue synchronization code to it ever, it will lead to potential deadlock